### PR TITLE
[export] Add a data type for representing export workflow information.

### PR DESCRIPTION
**Pull request by zhxchen17**
upstreaming some of our internal work to OSS so that we can get a better
preiew of how export pipeline works. there'll be more modularized work
sent in later.

Fixes #ISSUE_NUMBER

### CODE REVIEW

File: torch/_export/workflow.py [hunk 1]
```diff
@@ -0,0 +1,18 @@
+import dataclasses
+from typing import Callable, Tuple
+
+import torch
+
+@dataclasses.dataclass
+class ExportedProgram:
+    module: torch.fx.GraphModule
+    example_inputs: Tuple[torch.Tensor, ...]
+
```
Comment (tugsbayasgalan) [inline torch/_export/workflow.py:10]: Can we add field for in_spec, out_spec, mutation?

Comment (zhxchen17) [reply torch/_export/workflow.py:10]: I can add in_spec and out_spec here. 

Mutation is a bit controversial so I can add them later.

**Review by tugsbayasgalan — APPROVED** [review]

**Pull request closed without merging**